app/src/main/java/org/simple/clinic/patient/PatientSearchResult.kt

```diff
@@ -68,15 +68,18 @@ data class PatientSearchResult(
           LEFT JOIN PatientPhoneNumber PP ON PP.patientUuid = P.uuid
           LEFT JOIN (
         		SELECT BP.patientUuid, BP.createdAt, F.name facilityName, F.uuid facilityUuid
-        		FROM BloodPressureMeasurement BP
+        		FROM (
+                SELECT BP.patientUuid, BP.createdAt, BP.facilityUuid
+                FROM BloodPressureMeasurement BP
+                WHERE BP.deletedAt IS NULL
+                ORDER BY BP.createdAt DESC
+            ) BP
         		INNER JOIN Facility F ON BP.facilityUuid = F.uuid
-            GROUP BY BP.patientUuid
-        		ORDER BY BP.createdAt DESC
         	) BP ON (BP.patientUuid = P.uuid)
     """
     }
 
-    @Query("""$mainQuery WHERE P.uuid IN (:uuids) AND P.status = :status""")
+    @Query("""$mainQuery WHERE P.uuid IN (:uuids) AND P.status = :status GROUP BY P.uuid""")
```

Comment:
Oh, the other `searchByIds` is not used anymore since it was part of `searchV1`. I'll remove it.

Reply:
We changed the inner blood pressure measurement query to include a `WHERE` clause so that we could exclude deleted BPs. 

However, this broke the earlier group by because we cannot use a `WHERE` clause after the group by clause which is the case when we append the `mainQuery` in the actual dao method.